tools/examples/src/main.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { lstatSync, readdirSync } from 'fs';
+import { join } from 'path';
+import { bold, green } from 'chalk';
+import {
+  getExamplePackageMetadata,
+  ExamplePackageMetadata,
+  ExampleMetadata,
+} from './metadata';
+import { generateExamplesLibBarrelFile } from './generate-examples-lib-barrel';
+import { generateDemosAppExamplesModule } from './generate-examples-module';
+import { generateDemosAppRoutingModule } from './generate-routing-module';
+import { generateDemosAppNavItems } from './generate-nav-items';
+
+export const EXAMPLES_ROOT = join(
+  __dirname,
+  '../../../',
+  'libs',
+  'examples',
+  'src',
+);
+export const DEMOS_APP_ROOT = join(
+  __dirname,
+  '../../../',
+  'apps',
+  'demos',
+  'src',
+);
+
+/** Collect all files containing examples in the demos app. */
+async function getExamplesInPackages(): Promise<ExamplePackageMetadata[]> {
+  return (await Promise.all(
```

Comment:
please don't wrap awaits.

```typescript
const examples = readdirSync(EXAMPLES_ROOT)
      .map(name => join(EXAMPLES_ROOT, name))
      .filter(dir => lstatSync(dir).isDirectory())
      .map(dir => getExamplePackageMetadata(dir));
const result = await Promise.all(examples);
return result.filter(Boolean)

```

Reply:
Is there a reason why not to do it? Or just preferred code style

Reply:
way better readable – codestyle